File: 24-study/bfs/1697.py
import sys
input = sys.stdin.readline
from collections import deque

## 이동: x+1, x-1, x*2

n,k = map(int, input().split())

INF = float("inf")
MAX = 100000
dist = [INF for _ in range(MAX+1)]
dist[n]=0

def loc(x):
    location = [x+1, x-1, x*2]
    location = [v for v in location if 0<=v<=MAX]

    return location

def bfs(start):
    que = deque()
    que.append(start)
    while que:
        curr = que.popleft()
        for next_ in loc(curr):
            if dist[next_]>dist[curr]+1:
                dist[next_]=dist[curr]+1
                que.append(next_)

bfs(n)
print(dist[k])

File: 24-study/DFS:BFS/1697.js
// https://www.acmicpc.net/problem/1697
let fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs
  .readFileSync(filePath)
  .toString()
  .split(" ")
  .map((x) => Number(x));

let [n,k] = input
const MAX_ = 100000;
const INF = Number.MAX_SAFE_INTEGER;
const dist = Array.from({length:MAX_+1},(v,i)=>INF)


function loc(x){
  const location = [x+1, x-1, x*2]
  const res = location.filter((el) => 0 <= el && el <= MAX_);
  return res
}

function bfs(start){
  const que = [start]
  dist[start] = 0
  while (que.length>0){
    const curr = que.shift()
    for(const next_ of loc(curr)){
      if (dist[next_] > dist[curr] + 1) {
        dist[next_] = dist[curr] + 1;

        que.push(next_);
      }
    }
  }
}

bfs(n)
console.log(dist[k])